Fixing Issue #4 first try

Richardn2002 opened the pull request:
磨刀不误砍柴功

不过。。
第一遍编译通过我就满意了
要是还能玩我就惊了

Richardn2002 committed:
Fixing Issue #4 first try

磨刀不误砍柴功

Gennadiyev committed:
Update main.lua

Gennadiyev commented:
Start to compile the given codes!

Gennadiyev commented:
Compilation Success! Testing the application

Gennadiyev commented:
Issue found: **20 times click on "Change Mode" can be get repeatedly.

Gennadiyev commented:
Issue found: **Looks like a variable is used twice. Will look into further.**

Gennadiyev commented on code/main.lua:
```diff
@@ -116,42 +220,94 @@ test.onClick=function(test)
+		--Thens
+		--4
+		if cancelling < 5 then
+			print("哇！居然点了取消！真没想到啊！那我就陪你玩玩好了！把程序名字设定为鲲的英语名试试？")
```
`print("哇！居然点了取消！真没想到啊！那我就陪你玩玩好了！把程序名字设定为鲲的英语名试试"..string.rep("？",cancelling)) `

Gennadiyev replied:
This is to resolve Issue #7.

Richardn2002 replied:
Done

Gennadiyev commented on code/main.lua:
```diff
@@ -8,91 +8,195 @@ activity.setTitle('鲲')
+
+	--0
+	if achievementCounter == 0 and mode == 3 and adder2.text == "0" and error_catch.checked then
+		achievementCounter = achievementCounter + 1
```
`achievementCounter = 1` is a better choice. Optimized and clean. (Or you are thinking something evil...?)

Richardn2002 replied:
achievementCounter在条件满足时step+1，step==1触发，出发后清零，这样这个事件可以重复触发，没有什么问题
这样做是为了保持所有事件的结构一致

你写=1是因为你知道这个事件只需要1次条件符合就能触发，这很“特殊”，不“一般”

Richardn2002 replied:
Now using -1 to mark completion.Closing all related complaints below.

Gennadiyev commented on code/main.lua:
```diff
@@ -8,91 +8,195 @@ activity.setTitle('鲲')
+		achievementCounter = achievementCounter + 1
+	end
+	--1
+	if notChanging == 0 and mode == 0 and adder1.text == "20" and adder2.text == "10" then
```
Issue #12 Fixed here.

Richardn2002 replied:
I didn't mean to...so we can learn that formatting does good!

Gennadiyev commented on code/main.lua:
```diff
@@ -8,91 +8,195 @@ activity.setTitle('鲲')
+	end
+	--1
+	if notChanging == 0 and mode == 0 and adder1.text == "20" and adder2.text == "10" then
+		notChanging = notChanging + 1
```
`notChanging = 1`  please.

richard20020815 replied:
see above

Gennadiyev commented on code/main.lua:
```diff
@@ -8,91 +8,195 @@ activity.setTitle('鲲')
+		notChanging = notChanging + 1
+	end
+	--2
+	if hittingModes > 0 then
```
In this way every single click on "CALC" would reset the counter. Infinity +1! That's a fatal error!
Use the following line to replace:

`if hittingModes > 0 and hittingModes < 20 then`

Gennadiyev commented on code/main.lua:
```diff
@@ -8,91 +8,195 @@ activity.setTitle('鲲')
+		cancelling = 0
+	end
+	--6
+	if calcAndCalc < 50 then 
```
It just feels strange... I prefer to erase this `if` and make `calcAndCalc = calcAndCalc + 1` always executed. Only the `calcAndCalc == 50` will be granted the achievement right? I bet nobody could get it to inf...

Richardn2002 replied:
This is for formatting reasons.

Gennadiyev commented on code/main.lua:
```diff
@@ -8,91 +8,195 @@ activity.setTitle('鲲')
+		adder1.text="SIX"
+		运算符.text="MULTIPLIED BY"
+		adder2.text="NINE"
+		fortyTwo = fortyTwo + 1
```
Damn it. I want it to be `fortyTwo = 3`, please fix it.

Gennadiyev commented on code/main.lua:
```diff
@@ -8,91 +8,195 @@ activity.setTitle('鲲')
 		print("这是您第50次计算。那么，象征性给您成就半个！")
 		found=found+0.5
 		print("隐藏成就达成："..tostring(found).."/7")
+		calcAndCalc = calcAndCalc + 1 
```
`calcAndCalc = 51`. PLEASE!

Gennadiyev commented on code/main.lua:
```diff
@@ -8,91 +8,195 @@ activity.setTitle('鲲')
+	--0
+	if achievementCounter == 1 then
+		print("ERROR: DIVIDED BY 0")
+		print("隐藏成就计数器："..tostring(found).."/7。")
```
Your idea of the `ACHIEVEMENT_SUM` is great. Why not imply it?

Gennadiyev replied:
`print("隐藏成就计数器："..tostring(found).."/"..tostring(ACHIEVEMENT_SUM).."。") `

Gennadiyev replied:
Copy that for all the below please.

Richardn2002 replied:
Done.

Gennadiyev commented on code/main.lua:
```diff
@@ -8,91 +8,195 @@ activity.setTitle('鲲')
+	--1
+	if notChanging == 1 then
+		print("您就不能换一个数据嘛！我生气了！我要把答案搞成错误的！")
+		result.text="340"
```
`result.text=tostring(WHAT_KUN_WANTS) `

Richardn2002 replied:
Done.

Gennadiyev commented on code/main.lua:
```diff
@@ -8,91 +8,195 @@ activity.setTitle('鲲')
+		print("隐藏成就达成："..tostring(found).."/7")
+		adder1.text="SIX"
+		运算符.text="MULTIPLIED BY"
+		adder2.text="NINE"
```
While most people can't simply search `6*9` online and get relevant result in short time, I would prefer `The ultimate answer` still be implied. Try to find another way? Developers don't want their game being stuck by players haven't been reading a book the dev has read.

Richardn2002 replied:
我原来的意思是达成成就后变成6*9。。。可能我写错了

Gennadiyev replied:
Seems.

Richardn2002 replied:
There is no error.

Gennadiyev commented on code/main.lua:
```diff
@@ -8,91 +8,195 @@ activity.setTitle('鲲')
+		knowingWhatKunWants = knowingWhatKunWants + 1
+	end
+	if knowingWhatKunWants > 10 then
+		print("不给不给不给"..string.rep("！",knowingWhatKunWants))
```
But wait. The messages got way too long so easily, so I would prefer do this:
`print(string.rep("不给！",knowingWhatKunWants - 10)) `

Richardn2002 replied:
Done.

Gennadiyev commented on code/main.lua:
```diff
@@ -8,91 +8,195 @@ activity.setTitle('鲲')
+		hittingModes = hittingModes + 1
+	end
+	--4
+	if cancelling > 0 then
```
The `cancelling` achievement can be repeatedly acquired now. Fix this (similarly as the above).

Gennadiyev replied:
*Actually I do not expect the five 'Cancel's to be consecutive...*

Gennadiyev replied:
Oh yes, the corrected code:
`if cancelling > 0 and cancelling < 5 then`

Richardn2002 replied:
Done.

Richardn2002 replied:
All right.

Richardn2002 replied:
Done.

Gennadiyev commented on code/main.lua:
```diff
@@ -8,91 +8,195 @@ activity.setTitle('鲲')
+		fortyTwo = fortyTwo + 1
+	end
+	--4
+	if cancelling > 0 then
```
See line 169.

Gennadiyev commented on code/main.lua:
```diff
@@ -116,42 +220,94 @@ test.onClick=function(test)
+
+		--5
+		if appname.text=="鲲" and kunIsBack == 0 then
+			kunIsBack = kunIsBack + 1
```
I got too tired to complain... Okay it's your choice to ignore such lines.

Gennadiyev reviewed (changes requested):
Nice work. Almost impossible for me without a compiler beside me! Thanks a lot w~

richard20020815 commented:
@Gennadiyev Done

Richardn2002 commented:
My FastHub App has gone mad.Those "Done" comments are messed up.

Richardn2002 committed:
Clean Up Second Try

更加严格的格式
修改了fatal errors
完全发挥了ACHIEVEMENT_SUM 和 WHAT_KUN_WANTS
使用了-1标记成就完成
减少了鲲用！卖萌的强度

Richardn2002 commented:
oh no...there are some non-fatal bugs

I ctrl-vied too happily

Richardn2002 committed:
Clean Up Third Try

修改typo
加了一处碎碎念


解释格式：
所有ifs 用 > 判断是否符合条件
所有thens 用 == 判断结束

Richardn2002 commented:
Comon code...plz work

Gennadiyev commented on code/main.lua:
```diff
@@ -116,42 +238,94 @@ test.onClick=function(test)
+		--5
+		if kunIsBack == 1 then
+			found = found + 1
+			print("鲲又回来了")
```
Wait since when did this achievement rewards 1 pt.?

Richardn2002 committed:
Clean Up Fourth Try

0.5 points for kunIsBack
Modified fortyTwo
